storage/constraints.go

```diff
@@ -167,3 +168,28 @@ func parseSize(s string) (uint64, bool, error) {
 	}
 	return size, true, nil
 }
+
+// ToString returns a parsable string representation of the storage constraints.
+func ToString(c Constraints) (string, error) {
```

Comment:
Could this be a regular `Stringer` like `func (c Constraints) String() string`? Does it really need an error return (that is, will we ever actually end up with a constraint with all three fields empty)?

Reply:
There's nothing stopping someone from passing in an empty constraints struct
```
 c := storage.Constraints{}
 s:= c.String()
```
Or a constraints with a negative size etc. The issue is the fields on storage.Constraints are public so there's no stopping bad input. `ParseConstraints` guards against bad input, but we can still make the struct directly.
And my goal was to ensure the resulting string is guaranteed to be parsable. 
Whether this could happen in practice depends on how well other areas catch errors etc.

Reply:
Fair enough reasoning, thanks!